process/block/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -213,133 +175,46 @@ func (aap *alteredAccountsProvider) fetchTokensDataForMarkedAccount(
 	return nil
 }
 
-func (aap *alteredAccountsProvider) extractMoveBalanceAddresses(
+func (aap *alteredAccountsProvider) extractAddressesWithBalanceChange(
 	txPool *indexer.Pool,
 	markedAlteredAccounts map[string]*markedAlteredAccount,
 ) {
 	selfShardID := aap.shardCoordinator.SelfId()
-	for _, regularTx := range txPool.Txs {
-		senderShardID := aap.shardCoordinator.ComputeId(regularTx.GetSndAddr())
-		receiverShardID := aap.shardCoordinator.ComputeId(regularTx.GetRcvAddr())
-		if senderShardID != selfShardID && receiverShardID != selfShardID {
-			continue
-		}
+
+	aap.extractAddressesFromTxsHandlers(selfShardID, txPool.Txs, markedAlteredAccounts)
+	aap.extractAddressesFromTxsHandlers(selfShardID, txPool.Scrs, markedAlteredAccounts)
+	aap.extractAddressesFromTxsHandlers(selfShardID, txPool.Rewards, markedAlteredAccounts)
+	aap.extractAddressesFromTxsHandlers(selfShardID, txPool.Invalid, markedAlteredAccounts)
+}
+
+func (aap *alteredAccountsProvider) extractAddressesFromTxsHandlers(
+	selfShardID uint32,
+	txsHandlers map[string]data.TransactionHandler,
+	markedAlteredAccounts map[string]*markedAlteredAccount,
+) {
+	for _, txHandler := range txsHandlers {
+		senderShardID := aap.shardCoordinator.ComputeId(txHandler.GetSndAddr())
+		receiverShardID := aap.shardCoordinator.ComputeId(txHandler.GetRcvAddr())
```

Comment:
[optional]
Could extract sender & receiver address in separate variables, e.g.:
```go
senderAddr := txHandler.GetSndAddr()
receiverAddr :=txHandler.GetRcvAddr()
```

Reply:
done